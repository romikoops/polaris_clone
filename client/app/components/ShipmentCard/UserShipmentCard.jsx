import React, { Component } from 'react'
import { withNamespaces } from 'react-i18next'
import moment from 'moment'
import PropTypes from 'prop-types'
import { v4 } from 'uuid'
import styles from './ShipmentCard.scss'
import adminStyles from '../Admin/Admin.scss'
import {
  gradientTextGenerator,
  switchIcon,
  numberSpacing,
  splitName
} from '../../helpers'

class UserShipmentCard extends Component {
  constructor (props) {
    super(props)

    this.state = {}
  }

  handleView () {
    const { shipment, dispatches } = this.props
    dispatches.getShipment(shipment.id, true)
  }
  render () {
    const {
      shipment,
      theme,
      t
    } = this.props

    const gradientFontStyle =
      theme && theme.colors
        ? gradientTextGenerator(theme.colors.primary, theme.colors.secondary)
        : { color: 'black' }
    const deselectedStyle = {
      ...gradientTextGenerator('#DCDBDC', '#DCDBDC')
    }
    const plannedDate =
    shipment.has_pre_carriage ? shipment.planned_pickup_date : shipment.planned_origin_drop_off_date

    const destinationHubObj = splitName(shipment.destination_hub.name)
    const originHubObj = splitName(shipment.origin_hub.name)
    const timeRow = !!plannedDate && !!shipment.planned_etd && !!shipment.planned_eta
      ? (<div className={`layout-row flex-100 layout-align-start-center
    ${styles.middle_bottom_box} ${styles.smallText}`}
      >
        <div className="flex-20 layout-align-center-start">
          <span className="flex-100"><b>{t('common:pickupDate')}</b><br />
            <span className={`${styles.grey}`}>
              {moment(plannedDate).format('DD/MM/YYYY')}
            </span>
          </span>
        </div>
        <div className="flex-20 layout-align-center-start">
          <span className="flex-100"><b>{t('common:etd')}</b><br />
            <span className={`${styles.grey}`}>
              {moment(shipment.planned_etd).format('DD/MM/YYYY')}
            </span>
          </span>
        </div>
        <div className="flex-20 layout-align-center-start">
          <span className="flex-100"><b>{t('common:eta')}</b><br />
            <span className={`${styles.grey}`}>
              {moment(shipment.planned_eta).format('DD/MM/YYYY')}
            </span>
          </span>
        </div>
        <div className={`flex-40 layout-align-start-end ${styles.carriages}`}>
          <div className="layout-row layout-align-end-end">
            <i
              className={shipment.has_pre_carriage ? 'fa fa-check clip' : 'fa fa-times'}
              style={shipment.has_pre_carriage ? gradientFontStyle : { color: '#E0E0E0' }}
            />
            <p>{t('shipment:preCarriageBase')}</p>
          </div>
          <div className="layout-row layout-align-end-end">
            <i
              className={shipment.has_on_carriage ? 'fa fa-check clip' : 'fa fa-times'}
              style={shipment.has_on_carriage ? gradientFontStyle : { color: '#E0E0E0' }}
            />
            <p>{t('shipment:onCarriageBase')}</p>
          </div>
        </div>
      </div>) : (
        <div className={`layout-row flex-100 layout-align-start-center
      ${styles.middle_bottom_box} ${styles.smallText}`}
        >
          <div className="layout-row flex-100 layout-align-center-center">
            <i
              className={shipment.has_pre_carriage ? 'fa fa-check clip' : 'fa fa-times'}
              style={shipment.has_pre_carriage ? gradientFontStyle : { color: '#E0E0E0' }}
            />
            <p>{t('shipment:preCarriageBase')}</p>
          </div>
          <div className="layout-row flex-100 layout-align-center-center">
            <i
              className={shipment.has_on_carriage ? 'fa fa-check clip' : 'fa fa-times'}
              style={shipment.has_on_carriage ? gradientFontStyle : { color: '#E0E0E0' }}
            />
            <p>{t('shipment:onCarriageBase')}</p>
          </div>
        </div>
      )
    const cargoTypeString = shipment.load_type = 'container' ? t('cargo:container') : t('cargo:cargoItem')

    return (
      <div
        key={v4()}
        className={
          `flex-100 layout-align-start-stretch
          ${styles.container}`
        }
      >
        <div className={`layout-row flex-15 layout-align-center-center ${styles.topRight}`}>
          <p className={`${styles.check} pointy`}>{t('common:viewDetails')}</p>
        </div>
        <hr className={`flex-100 layout-row ${styles.hr_divider}`} />
        <div className={adminStyles.card_link} onClick={() => this.handleView()} />

        <div
          className={`layout-row flex-100 layout-align-space-between-stretch ${styles.top_box}`}
        >
          <div className={`layout-row flex-10 layout-align-center-center ${styles.routeIcon}`}>
            {switchIcon(shipment.mode_of_transport, gradientFontStyle)}
          </div>
          <div className={`flex-60 layout-row layout-align-start-center ${styles.hub_name}`}>
            <div className="layout-column layout-align-center-start">
              <p>{t('common:from')}:&nbsp;<span>{originHubObj.name}</span></p>
              <p>{t('common:to')}:&nbsp;<span>{destinationHubObj.name}</span></p>
            </div>
          </div>
          <div className={`layout-row flex-20 layout-align-start-center ${styles.ett}`}>
            {!!shipment.planned_etd && !!shipment.planned_eta ? (<div>
              <b>{moment(shipment.planned_eta).diff(shipment.planned_etd, 'days')} {t('common:days')}</b><br />
              <span className={`${styles.grey}`}>
                {t('shipment:estimatedTransitTime')}
              </span>
            </div>) : '' }
          </div>
        </div>
        <div
          className={`layout-row flex-100 layout-align-space-between-stretch ${styles.middle_top_box}`}
        >
          <div className="layout-row flex-35 layout-align-center-center">
            <div className="flex-100">
              <b className={styles.ref_row_card}>{t('common:ref')}:&nbsp;{shipment.imc_reference}</b>
              <p>{t('shipment:placedAt')}&nbsp;{moment(shipment.booking_placed_at).format('DD/MM/YYYY | HH:mm')}</p>
            </div>
          </div>

          <hr />

          <div className="layout-row flex-60 layout-align-center-center">
            <div className=" flex-100">
              <div className="layout-row flex-50 layout-align-start-center">
                <div className="flex-10 layout-row layout-align-center-center">
                  <i className="fa fa-user clip" style={gradientFontStyle} />
                </div>
                <div className="flex-80 layout-row layout-align-start-start">
                  <h4>{shipment.clientName}</h4>
                </div>
              </div>
              <div className="layout-row flex-50 layout-align-start-center">
                <span className="flex-10 layout-row layout-align-center-center">
                  <i className="fa fa-building clip" style={gradientFontStyle} />
                </span>
                <span className={`flex-80 layout-row layout-align-start-center ${styles.grey}`}>
                  <p>{shipment.companyName}</p>
                </span>
              </div>
            </div>
          </div>

        </div>

        {shipment.status !== 'finished' ? timeRow : (
          <div className={`layout-row flex-40 layout-align-start-stretch
            ${styles.middle_bottom_box} ${styles.smallText}`}
          >
            <div className="flex-100 layout-row"><b>{t('shipment:arrivedOn')}:&nbsp;</b>
              <span className={`${styles.grey}`}>
                {moment(shipment.planned_eta).format('DD/MM/YYYY')}
              </span>
            </div>
          </div>
        )}

        <div className={`layout-row flex-100 layout-align-space-between-center
            ${styles.bottom_box}`}
        >
          <div className="layout-row flex-65 layout-align-start-center">
            <div className="layout-row flex-10">
              <div className="layout-row layout-align-center-center">
                <span className={`${styles.smallText}`}>
                  <b>x</b><span className={`${styles.bigText}`}>{shipment.cargo_count}</span>
                </span>
              </div>
            </div>
            <span className="flex-35">{cargoTypeString}</span>
            <span className="flex-25 layout-row">
              <i
                className="fa fa-check-square clip"
                style={shipment.has_pre_carriage ? gradientFontStyle : deselectedStyle}
              />
              <p>{t('shipment:pickUp')}</p>
            </span>
            <span className="flex-25 layout row">
              <i
                className="fa fa-check-square clip"
                style={shipment.has_on_carriage ? gradientFontStyle : deselectedStyle}
              />
              <p>{t('shipment:delivery')}</p>
            </span>
          </div>
          <div className="layout-row flex layout-align-end-end">
            <span className={`${styles.bigText} ${styles.price_style}`}>
              <span>
                {shipment.edited_total
                  ? numberSpacing(shipment.edited_total.value, 2)
                  : numberSpacing(shipment.total_price.value, 2)}
              </span>
              <span>&nbsp;</span>
              <span> {shipment.total_price.currency} </span>
            </span>
          </div>
        </div>
      </div>
    )
  }
}

UserShipmentCard.propTypes = {
  shipment: PropTypes.objectOf(PropTypes.shipment),
  t: PropTypes.func.isRequired,
  dispatches: PropTypes.objectOf(PropTypes.func),
  theme: PropTypes.theme
}

UserShipmentCard.defaultProps = {
  shipment: {},
  dispatches: {},
  theme: {}
}

export default withNamespaces(['shipment', 'user', 'cargo'])(UserShipmentCard)
