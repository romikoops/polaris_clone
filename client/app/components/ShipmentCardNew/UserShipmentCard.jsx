import React, { Component } from 'react'
import moment from 'moment'
import PropTypes from 'prop-types'
import { v4 } from 'uuid'
import styles from './ShipmentCard.scss'
import GradientBorder from '../GradientBorder'
import {
  gradientTextGenerator,
  gradientGenerator,
  gradientBorderGenerator,
  switchIcon,
  totalPrice,
  formattedPriceValue
} from '../../helpers'

export class UserShipmentCard extends Component {
  constructor (props) {
    super(props)

    this.state = {}
  }

  handleView () {
    const { shipment, dispatches } = this.props
    dispatches.getShipment(shipment.id, true)
  }
  render () {
    const {
      shipment,
      theme,
      hubs
    } = this.props

    const gradientStyle =
      theme && theme.colors
        ? gradientGenerator(theme.colors.primary, theme.colors.secondary)
        : { background: 'black' }
    const gradientFontStyle =
      theme && theme.colors
        ? gradientTextGenerator(theme.colors.primary, theme.colors.secondary)
        : { color: 'black' }
    const deselectedStyle = {
      ...gradientTextGenerator('#DCDBDC', '#DCDBDC')
    }
    const gradientBorderStyle =
      theme && theme.colors
        ? gradientBorderGenerator(theme.colors.primary, theme.colors.secondary)
        : { background: 'black' }

    const bg1 =
      hubs.startHub && hubs.startHub.photo
        ? { backgroundImage: `url(${hubs.startHub.photo})` }
        : {
          backgroundImage:
            'url("https://assets.itsmycargo.com/assets/default_images/crane_sm.jpg")'
        }
    const bg2 =
      hubs.endHub && hubs.endHub.photo
        ? { backgroundImage: `url(${hubs.endHub.photo})` }
        : {
          backgroundImage:
            'url("https://assets.itsmycargo.com/assets/default_images/destination_sm.jpg")'
        }

    return (
      <div
        key={v4()}
        className={
          `layout-column flex-100 layout-align-start-stretch
          ${styles.container} ${styles.relative}`
        }
      >
        <div className={styles.card_link} onClick={() => this.handleView()} />
        <div className="layout-row layout-wrap flex-10 layout-wrap layout-align-center-center">
          <span className={`flex-100 ${styles.ref_row_card}`}>Ref: <b>{shipment.imc_reference}</b></span>
        </div>
        <div
          className={`layout-row flex-100 layout-align-space-between-stretch ${styles.section}`}
          onClick={() => this.handleView()}
        >
          <div className={`layout-row flex-55 layout-align-space-between-stretch ${styles.relative}`}>

            <GradientBorder
              wrapperClassName={`layout-column layout-align-end-stretch flex-50 ${styles.city} ${styles.margin_right}`}
              gradient={gradientBorderStyle}
              className="layout-column flex-100"
              content={(
                <div className="layout-column flex-100">
                  <div className="layout-align-center-center flex-100">
                    <div className={`flex-100 layout-align-center-center ${styles.hub_name}`}>
                      <p className="layout-align-center-center flex-100">{shipment.schedule_set['0'].origin_hub.name}</p>
                    </div>
                  </div>
                  <div className="layout-column flex-100">
                    <span className="flex-100" style={bg1} />
                  </div>
                </div>
              )}
            />

            <div className={`layout-row layout-align-center-center ${styles.routeIcon}`} style={gradientStyle}>
              {switchIcon(shipment.mode_of_transport)}
            </div>

            <GradientBorder
              wrapperClassName={`layout-column layout-align-end-stretch flex-50 ${styles.city} ${styles.margin_left}`}
              gradient={gradientBorderStyle}
              className="layout-column flex-100"
              content={(
                <div className="layout-column flex-100">
                  <div className={`flex-100 layout-align-center-start ${styles.hub_name}`}>
                    <p>{shipment.schedule_set['0'].destination_hub.name}</p>
                  </div>
                  <div className="layout-column flex-100">
                    <span className="flex-100" style={bg2} />
                  </div>
                </div>
              )}
            />

          </div>
          <div className={`layout-column flex-40 ${styles.user_info}`} onClick={() => this.handleView()}>
            <div className="layout-column flex-50 layout-align-center-stretch">
              <div className="layout-row flex-50 layout-align-start-start">
                <div className="flex-20 layout-row layout-align-center-center">
                  <i className="fa fa-user clip" style={gradientFontStyle} />
                </div>
                <div className="flex-80 layout-row layout-align-start-start">
                  <h4>{shipment.clientName}</h4>
                </div>
              </div>
              <div className="layout-row flex-50 layout-align-start-stretch">
                <span className="flex-20 layout-row layout-align-center-center">
                  <i className="fa fa-building clip" style={gradientFontStyle} />
                </span>
                <span className={`flex-80 layout-row layout-align-start-center ${styles.grey}`}>
                  <p>{shipment.companyName}</p>
                </span>
              </div>
            </div>
            <div className={`layout-row flex-50 layout-align-end-end ${styles.smallText}`}>
              <span className="flex-80"><b>Booking placed at</b><br />
                <span className={`${styles.grey}`}>
                  {moment(shipment.booking_placed_at).format('DD/MM/YYYY - HH:mm')}
                </span>
              </span>
            </div>
          </div>
        </div>
        <div className={`layout-row flex-40 layout-align-start-stretch
            ${styles.section} ${styles.separatorTop} ${styles.smallText}`}
        >
          <div className="layout-column flex-20">
            <span className="flex-100"><b>Pickup Date</b><br />
              <span className={`${styles.grey}`}>
                {moment(shipment.planned_pickup_date).format('DD/MM/YYYY')}
              </span>
            </span>
          </div>
          <div className="layout-column flex-20">
            <span className="flex-100"><b>ETD</b><br />
              <span className={`${styles.grey}`}>
                {moment(shipment.planned_etd).format('DD/MM/YYYY')}
              </span>
            </span>
          </div>
          <div className="layout-column flex-20">
            <span className="flex-100"><b>ETA</b><br />
              <span className={`${styles.grey}`}>
                {moment(shipment.planned_eta).format('DD/MM/YYYY')}
              </span>
            </span>
          </div>
          <div className="layout-column flex-40">
            <span className="flex-100"><b>Estimated Transit Time</b><br />
              <span className={`${styles.grey}`}>
                {moment(shipment.planned_eta).diff(shipment.planned_etd, 'days')} days
              </span>
            </span>
          </div>
        </div>
        <div className={`layout-row flex-25 layout-align-space-between-center
            ${styles.sectionBottom} ${styles.separatorTop}`}
        >
          <div className="layout-row flex-60 layout-align-start-center">
            <div className="layout-row flex-15">
              <div className={`layout-row layout-align-center-center ${styles.green_icon}`} style={gradientStyle}>
                <span className={`${styles.smallText}`}>
                  <b>x</b><span className={`${styles.bigText}`}>1</span>
                </span>
              </div>
            </div>
            <span className="flex-30">Cargo item</span>
            <span className="flex-30">
              <i
                className="fa fa-check-square clip"
                style={shipment.pickup_address ? gradientFontStyle : deselectedStyle}
              />
              <span> pickup</span>
            </span>
            <span className="flex-30">
              <i
                className="fa fa-check-square clip"
                style={shipment.delivery_address ? gradientFontStyle : deselectedStyle}
              />
              <span> delivery</span>
            </span>
          </div>
          <div className="layout-align-end-center">
            <span className={`${styles.bigText}`}>
              <span>{totalPrice(shipment).currency} </span>
              <span>
                {formattedPriceValue(totalPrice(shipment).value)}
              </span>
            </span>
          </div>
        </div>
      </div>
    )
  }
}

UserShipmentCard.propTypes = {
  shipment: PropTypes.objectOf(PropTypes.shipment),
  dispatches: PropTypes.objectOf(PropTypes.func),
  theme: PropTypes.theme,
  hubs: PropTypes.objectOf(PropTypes.hub)
}

UserShipmentCard.defaultProps = {
  shipment: {},
  dispatches: {},
  theme: {},
  hubs: {}
}

export default UserShipmentCard
