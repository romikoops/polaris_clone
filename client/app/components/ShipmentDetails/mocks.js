// Will remove it later
// ============================================
import { get } from 'lodash'

export const trucking = { preCarriage: { truckType: '' }, onCarriage: { truckType: '' } }
export const id = 4606
export const selectedDay = '2019-01-24T10:00:00.000Z'

export const ShipmentDetailsAvailableRoutes = [{
  itineraryId: 2863,
  itineraryName: 'Gothenburg - Qingdao',
  modeOfTransport: 'ocean',
  origin: {
    stopId: 4656, hubId: 3023, hubName: 'Gothenburg Port', nexusId: 597, nexusName: 'Gothenburg', latitude: 57.694253, longitude: 11.854048, country: 'SE', truckTypes: ['default']
  },
  destination: {
    stopId: 4657, hubId: 3027, hubName: 'Qingdao Port', nexusId: 601, nexusName: 'Qingdao', latitude: 36.083811, longitude: 120.323534, country: 'CN', truckTypes: []
  }
}, {
  itineraryId: 2958,
  itineraryName: 'Gothenburg - Qingdao',
  modeOfTransport: 'air',
  origin: {
    stopId: 4846, hubId: 3030, hubName: 'Gothenburg Airport', nexusId: 597, nexusName: 'Gothenburg', latitude: 57.694253, longitude: 11.854048, country: 'SE', truckTypes: ['default']
  },
  destination: {
    stopId: 4847, hubId: 3034, hubName: 'Qingdao Airport', nexusId: 601, nexusName: 'Qingdao', latitude: 36.083811, longitude: 120.323534, country: 'CN', truckTypes: []
  }
}]

export const ShipmentDetails = {
  availableRoutes: ShipmentDetailsAvailableRoutes,
  availableMots: ['ocean', 'air']
}

export const cargoItemTypes = [{
  id: 25, dimension_x: null, dimension_y: null, description: 'Pallet', area: null, created_at: '2018-06-27T17:28:28.431Z', updated_at: '2018-06-27T17:28:28.431Z', category: 'Pallet'
}, {
  id: 22, dimension_x: null, dimension_y: null, description: 'Carton', area: null, created_at: '2018-06-27T17:28:28.417Z', updated_at: '2018-06-27T17:28:28.417Z', category: 'Carton'
}, {
  id: 23, dimension_x: null, dimension_y: null, description: 'Crate', area: null, created_at: '2018-06-27T17:28:28.422Z', updated_at: '2018-06-27T17:28:28.422Z', category: 'Crate'
}, {
  id: 26, dimension_x: null, dimension_y: null, description: 'Bottle', area: null, created_at: '2018-06-27T17:28:28.436Z', updated_at: '2018-06-27T17:28:28.436Z', category: 'Bottle'
}, {
  id: 27, dimension_x: null, dimension_y: null, description: 'Stack', area: null, created_at: '2018-06-27T17:28:28.440Z', updated_at: '2018-06-27T17:28:28.440Z', category: 'Stack'
}, {
  id: 28, dimension_x: null, dimension_y: null, description: 'Drum', area: null, created_at: '2018-06-27T17:28:28.445Z', updated_at: '2018-06-27T17:28:28.445Z', category: 'Drum'
}, {
  id: 29, dimension_x: null, dimension_y: null, description: 'Skid', area: null, created_at: '2018-06-27T17:28:28.450Z', updated_at: '2018-06-27T17:28:28.450Z', category: 'Skid'
}, {
  id: 30, dimension_x: null, dimension_y: null, description: 'Barrel', area: null, created_at: '2018-06-27T17:28:28.454Z', updated_at: '2018-06-27T17:28:28.454Z', category: 'Barrel'
}]

export const maxDimensions = {
  general: {
    dimensionX: '590.0', dimensionY: '234.2', dimensionZ: '228.0', payloadInKg: '21770.0', chargeableWeight: '21770.0'
  },
  air: {
    dimensionX: '120.0', dimensionY: '100.0', dimensionZ: '150.0', payloadInKg: '1000.0', chargeableWeight: '1000.0'
  }
}

export const scope = {
  links: { about: '', legal: '' },
  terms: ['You verify that all the information provided above is true', 'You agree to the presented terms and conditions.', 'Demo is to discuss the validity of the presented prices with the product owners.'],
  fee_detail: 'key_and_name',
  closed_shop: false,
  has_customs: true,
  has_insurance: true,
  fixed_currency: false,
  dangerous_goods: false,
  cargo_info_level: 'hs_codes',
  carriage_options: { on_carriage: { export: 'optional', import: 'optional' }, pre_carriage: { export: 'optional', import: 'optional' } },
  detailed_billing: false,
  total_dimensions: true,
  consolidate_cargo: false,
  modes_of_transport: {
    air: { container: true, cargo_item: true }, rail: { container: true, cargo_item: true }, ocean: { container: true, cargo_item: true }, truck: { container: false, cargo_item: false }
  },
  show_beta_features: true,
  closed_registration: false,
  continuous_rounding: false,
  incoterm_info_level: 'text',
  non_stackable_goods: true,
  open_quotation_tool: false,
  customs_export_paper: false,
  fixed_exchange_rates: false,
  require_full_address: true,
  closed_quotation_tool: false
}

export const theme = {
  colors: {
    primary: '#0D5BA9', secondary: '#23802A', brightPrimary: '#2491FD', brightSecondary: '#25ED36'
  },
  logoWide: 'https://assets.itsmycargo.com/assets/logos/Logo_transparent_blue.png',
  logoLarge: 'https://assets.itsmycargo.com/assets/logos/logo_box.png',
  logoSmall: 'https://assets.itsmycargo.com/assets/logos/logo_box.png',
  logoWhite: 'https://assets.itsmycargo.com/assets/logos/Logo_transparent_white.png',
  background: 'https://assets.itsmycargo.com/assets/images/cropped_banner_2.jpg'
}

export const origin = {
  latitude: 57.694253, longitude: 11.854048, nexusId: 597, nexusName: 'Gothenburg', country: 'SE'
}
export const destination = {
  latitude: 36.083811, longitude: 120.323534, nexusId: 601, nexusName: 'Qingdao', country: 'CN'
}
export const cargoItem = {
  payloadInKg: 11, totalVolume: 0, totalWeight: 0, dimensionX: 7, dimensionY: 7, dimensionZ: 8, quantity: 7, cargoItemTypeId: 23, dangerousGoods: false, stackable: false
}

export const cargoItemAggregated = {
  payloadInKg: 0, totalVolume: 122, totalWeight: 346, dimensionX: 0, dimensionY: 0, dimensionZ: 0, quantity: 1, cargoItemTypeId: '', dangerousGoods: false, stackable: true
}

export const cargoItemContainer = {
  sizeClass: 'highCube', quantity: 14, dangerousGoods: false, weight: 16
}

export const cargoUnits = [cargoItem]

export const identity = x => x

export const importedProps = {
  ShipmentDetails,
  aggregatedCargo: false,
  cargoItemTypes,
  cargoUnits,
  destination,
  direction: 'export',
  id,
  loadType: 'cargo_item',
  maxDimensions,
  onCarriage: false,
  onChangeCargoUnitCheckbox: identity,
  onChangeCargoUnitInput: identity,
  onChangeCargoUnitSelect: identity,
  onDeleteUnit: identity,
  onUpdateCargoUnit: identity,
  origin,
  preCarriage: false,
  scope,
  selectedDay,
  theme,
  toggleModal: identity,
  trucking
}

export function logPath (props, path) {
  console.log(path, get(props, path, 'NO SUCH PATH'))
}

export function log (props) {
  let holder = '{\n'
  const declarations = []
  Object.keys(props).forEach((key) => {
    const toLog = `const ${key} = ${JSON.stringify(props[key])}`
    if (toLog.length < 150) console.log(toLog)
    declarations.push(toLog)

    holder = `${holder}\n  ${key},`
  })
  holder = `${holder}\n}`
  console.log(declarations.join('\n'))
  // console.log(holder)
  // console.log(JSON.stringify(props, null, 2))
}
