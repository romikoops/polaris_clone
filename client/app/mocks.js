import { set, cloneDeep } from 'lodash'

/**
 * Checks if variable is an non-empty object
 *
 * True for `{a:1}`
 * Fase for `null, {}, [], [1,2,3]`
 */
const isObject = (x) => {
  const ok = x !== null && !Array.isArray(x) && typeof x === 'object'
  if (!ok) {
    return false
  }

  return Object.keys(x).length > 0
}
/* eslint-disable */
/**
 * Used in unit test to modify specific properties
 * with minimal lines of code
 *
 * You can specify `path` to be empty string
 * if the change affects more than one of object's branches
 */
export const change = (origin, pathRaw, rules) => {
  const willReturn = cloneDeep(origin)
  
  if (!isObject(rules)) {
    set(willReturn, pathRaw, rules)
    return willReturn
  }
  const path = pathRaw === '' ? '' : `${pathRaw}.`
  for (const ruleKey of Object.keys(rules)) {
    const rule = rules[ruleKey]
    if (!isObject(rule)) {
      set(
        willReturn,
        `${path}${ruleKey}`,
        rule
      )
      continue
    }
    Object.keys(rule).filter(subruleKey => !isObject(rule[subruleKey])).map((subruleKey) => {
      const subrule = rule[subruleKey]
      set(
        willReturn,
        `${path}${ruleKey}.${subruleKey}`,
        subrule
      )
    })
    Object.keys(rule).filter(subruleKey => isObject(rule[subruleKey])).map((subruleKey) => {
      const subrule = rule[subruleKey]
      Object.keys(subrule).map((deepKey) => {
        const deep = rule[subruleKey][deepKey]
        set(
          willReturn,
          `${path}${ruleKey}.${subruleKey}.${deepKey}`,
          deep
        )
      })
    })
  }

  return willReturn
}
/* eslint-enable */
export const identity = input => input

export const theme = {
  colors: {
    primary: '#333',
    secondary: '#fafafa'
  }
}

export const user = {
  role: { name: 'shipper' },
  company_name: 'FOO_COMPANY',
  currency: 'EUR',
  email: 'foo@bar.baz',
  first_name: 'John',
  guest: false,
  last_name: 'Doe',
  phone: '6345789'
}

export const history = {
  push: identity
}

export const tenant = {
  data: {
    id: 123,
    scope: {
      modes_of_transport: {
        ocean: {
          OCEAN_LOAD_TYPE: true
        },
        air: {},
        truck: {},
        rail: {}
      }
    },
    theme,
    subdomain: 'foosubdomain'
  }
}

export const req = {
  schedule: [{}],
  total: 345,
  shipment: {}
}

const schedulesInShipmentData = [
  { hub_route_key: 'OCEAN_HUB_ROUTE_KEY', mode_of_transport: 'ocean' },
  { hub_route_key: 'AIR_HUB_ROUTE_KEY', mode_of_transport: 'air' },
  { hub_route_key: 'TRUCK_HUB_ROUTE_KEY', mode_of_transport: 'truck' },
  { hub_route_key: 'RAIL_HUB_ROUTE_KEY', mode_of_transport: 'rail' }
]

export const shipmentInShipmentData = {
  load_type: 'OCEAN_LOAD_TYPE',
  selected_offer: { cargo: {}, total: { value: 77 } },
  schedules_charges: {
    OCEAN_HUB_ROUTE_KEY: { total: 40 },
    AIR_HUB_ROUTE_KEY: { total: 200 },
    TRUCK_HUB_ROUTE_KEY: { total: 75 },
    RAIL_HUB_ROUTE_KEY: { total: 125 }
  },
  total_goods_value: {
    value: 15,
    currency: 'USD'
  },
  trucking: {
    pre_carriage: { trucking_time_in_seconds: 55 }
  },
  has_on_carriage: false,
  has_pre_carriage: false,
  notes: 'FOO_NOTES',
  cargo_notes: 'FOO_CARGO_NOTES',
  eori: 1234,
  incoterm_text: 'FOO_INCOTERM_TEXT',
  total_price: {
    value: 12,
    currency: 'USD'
  }
}

export const shipmentData = {
  contacts: [],
  shipment: shipmentInShipmentData,
  documents: [],
  cargoItems: [],
  containers: [],
  schedules: schedulesInShipmentData,
  locations: {
    startHub: 'FOO_START_HUB',
    endHub: 'FOO_END_HUB',
    destination: {
      street_number: 9,
      street: 'BAR_STREET',
      city: 'BAR_CITY',
      country: 'China',
      zip_code: 845321
    },
    origin: {
      street_number: 7,
      street: 'FOO_STREET',
      city: 'FOO_CITY',
      country: 'Germany',
      zip_code: 21177
    }
  }
}

export const match = {
  params: {}
}

export const shipments = {
  direction: 'export',
  finished: [],
  open: [],
  requested: []
}

export const shipment = {
  origin_hub: { name: 'FOO_ORIGIN_HUB' },
  destination_hub: { name: 'FOO_DESTINATION_HUB' },
  id: 654,
  status: 'FOO_STATUS',
  clientName: 'FOO_CLIENT_NAME',
  planned_etd: 789,
  imc_reference: 'FOO_IMC_REFERENCE',
  schedule_set: []
}

export const location = {
  primary: true,
  id: 246,
  street_number: 579,
  streetNumber: 579,
  geocodedAddress: 'FOO_GEOCODED_ADDRESS',
  street: 'FOO_STREET',
  city: 'FOO_CITY',
  zip_code: '22456',
  zipCode: '22456',
  country: 'Germany'
}

export const address = {
  id: 13579,
  street_number: '1',
  street: 'Uhlandweg',
  city: 'Hamburg',
  zip_code: '22848',
  country: 'Germany'
}

export const hub = {
  location,
  name: 'FOO_HUB_NAME'
}

export const client = {
  first_name: 'Sebastian',
  last_name: 'Muller',
  email: 'muller55@yahoo.de',
  company_name: 'Mehristweniger',
  password: 'pwd'
}

export const route = {
  id: 24680
}

export const vehicleType = {
  id: 827364,
  name: 'FOO_VEHICLE_TYPE_NAME',
  mode_of_transport: 'FOO_VEHICLE_TYPE_TRANSPORT'
}

export const schedule = {
  hub_route_key: 'FOO_SCHEDULE_ROUTE_KEY',
  id: 555777,
  eta: 441188
}

export const charge = {
  id: 654297
}

export const locations = {
  origin: {},
  destination: {}
}

class MapMock {
  constructor (x) {
    this.x = x
  }
  bindTo () {
    return this.x
  }
  setContent () {
    return this.x
  }
  addListener () {
    return this.x
  }
}

export const gMaps = {
  InfoWindow: MapMock,
  LatLngBounds: MapMock,
  Map: MapMock,
  MapTypeId: { ROADMAP: '' },
  Marker: MapMock,
  Point: MapMock,
  Size: MapMock,
  places: { Autocomplete: MapMock }
}
