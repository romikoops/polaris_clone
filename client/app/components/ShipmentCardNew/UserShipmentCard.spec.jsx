import * as React from 'react'
import { shallow } from 'enzyme'
import { shipment } from '../../mocks'

import { UserShipmentCard } from './UserShipmentCard'

jest.mock('moment', () => {
  const format = () => 19

  return () => ({
    format
  })
})

const propsBase = {
  shipment
}

test('shallow rendering', () => {
  expect(shallow(<UserShipmentCard {...propsBase} />)).toMatchSnapshot()
})
