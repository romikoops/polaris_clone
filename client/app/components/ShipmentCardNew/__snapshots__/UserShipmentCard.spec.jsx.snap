// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow rendering 1`] = `
<div
  className="layout-column flex-100 layout-align-start-stretch
          container relative"
>
  <div
    className="layout-row layout-align-space-around-center topRightBox topRightColor"
  >
    <b>
      REQUESTED
    </b>
  </div>
  <div
    className="layout-row layout-wrap flex-10 layout-wrap layout-align-center-center"
  >
    <span
      className="flex-100 title"
    >
      Ref: 
      <b>
        FOO_IMC_REFERENCE
      </b>
    </span>
  </div>
  <div
    className="layout-row flex-90 layout-align-space-between-stretch section"
  >
    <div
      className="layout-row flex-50 layout-align-space-between-stretch
              relative"
    >
      <div
        className="layout-column flex-45 city"
      >
        <div
          className="layout-column layout-padding flex-50 layout-align-center-start"
        >
          <span>
            <br />
          </span>
        </div>
        <div
          className="layout-column flex-50"
        >
          <img
            className="flex-100"
            src="/app/assets/images/dashboard/stockholm.png"
          />
        </div>
      </div>
      <div
        className="layout-row layout-align-center-center routeIcon"
      >
        <i
          className="fa fa-ship"
        />
      </div>
      <div
        className="layout-column flex-45 city"
      >
        <div
          className="layout-column layout-padding flex-50 layout-align-center-start"
        >
          <span>
            <br />
          </span>
        </div>
        <div
          className="layout-column flex-50"
        >
          <img
            className="flex-100"
            src="/app/assets/images/dashboard/shanghai.png"
          />
        </div>
      </div>
    </div>
    <div
      className="layout-column flex-40"
    >
      <div
        className="layout-column flex-50 layout-align-center-stretch"
      >
        <div
          className="layout-row flex-50 layout-align-start-stretch"
        >
          <div
            className="flex-20"
          >
            <i
              className="fa fa-user profileIcon"
            />
          </div>
          <div
            className="flex-80"
          >
            FOO_CLIENT_NAME
          </div>
        </div>
        <div
          className="layout-row flex-50 layout-align-start-stretch"
        >
          <span
            className="flex-20"
          >
            <i
              className="fa fa-building profileIcon"
            />
          </span>
          <span
            className="flex-80 grey"
          />
        </div>
      </div>
      <div
        className="layout-row flex-50 layout-align-end-end smallText"
      >
        <span
          className="flex-80"
        >
          <b>
            Booking placed at
          </b>
          <br />
          <span
            className="grey"
          >
            19
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
